packages/common/src/store/ui/buy-sell/swapFormSchema.ts

```diff
@@ -22,19 +14,43 @@ export const createSwapFormSchema = (
   z.object({
     inputAmount: z
       .string()
-      .min(1, { message: messages.inputAmountRequired })
-      .refine((val) => !isNaN(parseFloat(val)), {
-        message: messages.invalidAmount
-      })
-      .refine((val) => parseFloat(val) >= min, {
-        message: messages.minAmount(min, tokenSymbol)
-      })
-      .refine((val) => parseFloat(val) <= max, {
-        message: messages.maxAmount(max, tokenSymbol)
-      })
-      .refine((val) => !balance || parseFloat(val) <= balance, {
-        message: messages.insufficientBalance(tokenSymbol)
-      }),
+      .refine(
+        (val) => {
+          // Allow empty string during typing, but require it for final validation
+          if (val === '') return false
+          return !isNaN(parseFloat(val))
+        },
+        {
+          message: messages.emptyAmount
+        }
+      )
+      .refine(
+        (val) => {
+          if (val === '') return false
```

Comment:
do we really need the empty string check at every refine step?

Reply:
yea helps not show errors falsely when stuff is processing